src/main/java/nl/tudelft/dnainator/ui/widgets/animations/SlidingAnimation.java

```diff
@@ -1,99 +1,108 @@
 
-import javafx.animation.Animation;
-import javafx.animation.Transition;
 import javafx.scene.layout.Pane;
 import javafx.util.Duration;
 
 /**
- * This class realises a sliding animation, to show and hide a pane and its contents.
- * TODO: extend for multiple directions.
+ * This class realises a sliding animation.
+ * It may slide up, down left or right.
  */
-public class SlidingAnimation {
-//	/**
-//	 * An enum that specifies the location of the {@link Pane} to be animated. Using this
-//	 * location, the translations can be set correctly.
-//	 * TODO: add more directions.
-//	 */
-//	public enum Location {
-//		TOP_LEFT
-//	}
-
-	private Pane pane;
-//	private Location location;
-	private Animation show;
-	private Animation hide;
-	private int width;
-	private int duration;
-
+public class SlidingAnimation extends TransitionAnimation {
+	
+	private double curWidth;
+	private double curHeight;
+	
 	/**
 	 * Creates the sliding animation, using the provided parameters for the duration, amount to
 	 * slide, and the location of the animation.
 	 *
 	 * @param pane         The {@link Pane} to be animated.
-	 * @param width        The length over which the pane will slide.
+	 * @param width        The horizontal length over which the pane will slide.
+	 * @param height       The vertical length over which the pane will slide.
 	 * @param duration     The duration of the animations.
-//	  param paneLocation The {@link SlidingAnimation.Location} of the {@link Pane}.
+	 * @param direction    The {@link TransitionAnimation.Direction} of the sliding animation.
 	 */
-	public SlidingAnimation(Pane pane, int width, int duration/*, Location paneLocation*/) {
-		this.pane = pane;
-		this.width = width;
-		this.duration = duration;
-//		this.location = paneLocation;
-
-		setupShowAnimation();
-		setupHideAnimation();
+	public SlidingAnimation(Pane pane, int width, int height, int duration, Direction direction) {
+		super(pane, width, height, duration, direction);
 	}
-
+	
 	/**
-	 * Hides the pane in a sliding fashion.
+	 * Creates a sliding animation with default {@link Direction#LEFT}.
+	 * @param pane         The {@link Pane} to be animated.
+	 * @param width        The length over which the pane will slide.
+	 * @param height       The vertical length over which the pane will slide.
+	 * @param duration     The duration of the animations.
 	 */
-	private void setupHideAnimation() {
-		hide = new Transition() {
-			{
-				setCycleDuration(Duration.millis(duration));
-			}
+	public SlidingAnimation(Pane pane, int width, int height, int duration) {
+		this(pane, width, height, duration, Direction.LEFT);
+	}
+	
+	
+	@Override
+	public void setupAnimation() {
+		curWidth = width;
+		curHeight = height;
+		setCycleDuration(Duration.millis(duration));
+	}
 
-			@Override
-			protected void interpolate(double frac) {
-				double curWidth = width * (1.0 - frac);
+	@Override
+	protected void interpolate(double frac) {
+		switch (direction) {
```

Comment:
Consider using dispatch, instead of a switch statement. For example, create a `LeftSlidingAnimation`, which returns a `RightSlidingAnimation` upon calling a method `opposite`. Then, in the toggle function you can update the animation and then `play` it